### PR TITLE
Add getWinnerId function, refactor testing

### PR DESCRIPTION
**Pull request by vitsalis**

### CODE REVIEW

File: models/round.php [hunk 1]
```diff
@@ -58,16 +58,32 @@ public function __construct( $a = false, $b = false ) {
...
         }
+        public function getWinnerId() {
+            if ( !$this->isFinalRound() ) {
+                throw new ModelValidationException( 'There is no winner if the game is not over' );
+            }
+            $usersAlive = $this->getUsersAlive();
+            if ( !empty( $usersAlive ) ) {
+                return end( $usersAlive )->id;
+            }
+            else {
```
Comment (dionyziz) [inline models/round.php]: Avoid `else` after `return`.

File: tests/models/round.php [hunk 2]
```diff
@@ -40,28 +40,62 @@ public function testError() {
...
+            $users = [ $user1->id => $user1, $user2->id => $user2 ];
+            $round = $this->buildRoundWithCreatures( $users );
+            $caught = false;
+            try {
+                $round->getWinnerId();
+            }
+            catch ( ModelValidationException $e ) {
+                $caught = true;
+            }
+            $this->assertTrue( $caught, 'A ModelValidationException must be caught if tha round is not the final one' );
```
Comment (dionyziz) [inline tests/models/round.php]: "the"?

File: models/round.php [hunk 3]
```diff
@@ -58,16 +58,32 @@ public function __construct( $a = false, $b = false ) {
...
+                return 0;
```
Comment (dionyziz) [inline models/round.php]: Prefer `false` or an exception to indicate an exceptional condition.

File: models/round.php [hunk 4]
```diff
@@ -58,16 +58,30 @@ public function __construct( $a = false, $b = false ) {
...
         }
+        public function getWinnerId() {
+            if ( !$this->isFinalRound() ) {
+                throw new ModelValidationException( 'There is no winner if the game is not over' );
+            }
+            $usersAlive = $this->getUsersAlive();
+            if ( !empty( $usersAlive ) ) {
+                return end( $usersAlive )->id;
+            }
+            false;
```
Comment (dionyziz) [inline models/round.php]: PHP isn't Scheme.

File: tests/models/round.php [hunk 5]
```diff
@@ -40,28 +40,62 @@ public function testError() {
...
+            $users = [ $user1->id => $user1, $user2->id => $user2 ];
+            $round = $this->buildRoundWithCreatures( $users );
+            $caught = false;
+            try {
+                $round->getWinnerId();
+            }
+            catch ( ModelValidationException $e ) {
+                $caught = true;
+            }
+            $this->assertTrue( $caught, 'A ModelValidationException must be caught if the round is not the final one' );
```
Comment (dionyziz) [inline tests/models/round.php]: Use `assertThrows`

Comment (vitsalis) [inline tests/models/round.php]: I don't know how to give the function as a parameter..

Comment (dionyziz) [inline tests/models/round.php]: See how it's done in other tests

**Pull request merged**